### DISCUSSION TITLE
Can't overwrite actively downloaded file

### DISCUSSION THREAD

**Discussion by VVvKamper** [minio/minio#11602] [Q&A] [answered]
We're trying to use minio as cdn server. We deployed it by minio operator (3.0.29) in distributed mode (4 servers).
We have a lot of people downloading from the minio (every moment there is at least one active download) and
seems like because of that we can't overwrite object that is actively downloaded. If we try to put new object everything works
fine or if we try to overwrite existing one that not actively downloading. We enabled versioning but it didn't help.
Can't find anything related in the docs or in the github issues.
To sum up - we can't overwrite object that are actively downloaded even with enabled bucket versioning. We are using latest minio release 2021.02.19.

**Comment by harshavardhana** [comment, marked as answer]
Of course, you can't overwrite an object that is being read - that would lead to inconsistent behavior - also versioning works at object level you cannot update a version while reading from the object version.

**Reply by VVvKamper** [reply]
i want to upload new version and want new reads to be from this new version. it sounds consistent for me. i'm not sure this is theoretical possible in every case, but as i understand s3 allows such scenario.

Right now i need to use new object and change external system (for example website) to look at this new path. If it works as intended it's ok - it just means minio not fit for this particular use case :(

**Reply by harshavardhana** [reply]
It's not possible - we are a strongly consistent system, we do not allow such type of behavior.  Please choose a new object.

**Reply by VVvKamper** [reply]
Ok, thanks a lot for clarification. I think i can make simple wrapper with needed behaviour on top of the minio.
